clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1020,34 +1020,38 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
             MPM.addPass(InstrProfiling(*Options, false));
           });
 
-    if (CodeGenOpts.OptimizationLevel == 0) {
-      MPM = PB.buildO0DefaultPipeline(Level, IsLTO || IsThinLTO);
-    } else if (IsThinLTO) {
-      MPM = PB.buildThinLTOPreLinkDefaultPipeline(Level);
-    } else if (IsLTO) {
-      MPM = PB.buildLTOPreLinkDefaultPipeline(Level);
-    } else {
-      MPM = PB.buildPerModuleDefaultPipeline(Level);
+    if (!CodeGenOpts.DisableSYCLEarlyOpts) {
+      if (CodeGenOpts.OptimizationLevel == 0) {
+        MPM = PB.buildO0DefaultPipeline(Level, IsLTO || IsThinLTO);
+      } else if (IsThinLTO) {
+        MPM = PB.buildThinLTOPreLinkDefaultPipeline(Level);
+      } else if (IsLTO) {
+        MPM = PB.buildLTOPreLinkDefaultPipeline(Level);
+      } else {
+        MPM = PB.buildPerModuleDefaultPipeline(Level);
+      }
     }
 
     if (!CodeGenOpts.MemoryProfileOutput.empty()) {
       MPM.addPass(createModuleToFunctionPassAdaptor(MemProfilerPass()));
       MPM.addPass(ModuleMemProfilerPass());
     }
-  }
-  if (LangOpts.SYCLIsDevice) {
-    MPM.addPass(SYCLMutatePrintfAddrspacePass());
-    if (!CodeGenOpts.DisableLLVMPasses && LangOpts.EnableDAEInSpirKernels)
-      MPM.addPass(DeadArgumentEliminationSYCLPass());
-  }
 
-  // Add SPIRITTAnnotations pass to the pass manager if
-  // -fsycl-instrument-device-code option was passed. This option can be used
-  // only with spir triple.
-  if (LangOpts.SYCLIsDevice && CodeGenOpts.SPIRITTAnnotations) {
-    assert(TargetTriple.isSPIR() &&
-           "ITT annotations can only be added to a module with spir target");
-    MPM.addPass(SPIRITTAnnotationsPass());
+    if (LangOpts.SYCLIsDevice) {
+      MPM.addPass(SYCLMutatePrintfAddrspacePass());
+      if (LangOpts.EnableDAEInSpirKernels)
+        MPM.addPass(DeadArgumentEliminationSYCLPass());
+    }
+
+    // Add SPIRITTAnnotations pass to the pass manager if
+    // -fsycl-instrument-device-code option was passed. This option can be used
+    // only with spir triple.
+    if (LangOpts.SYCLIsDevice && CodeGenOpts.SPIRITTAnnotations &&
+        !CodeGenOpts.DisableSYCLEarlyOpts) {
```

Comment:
This is not an optimization pass as well, it should be scheduled regardless of `DisableSYCLEarlyOpts`